app/src/main/java/net/aiscope/gdd_app/ui/mask/MaskActivity.kt

```diff
@@ -104,6 +109,45 @@ class MaskActivity : AppCompatActivity(), MaskView, CaptureFlow {
 
     }
 
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.mask_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when(item.itemId) {
+            R.id.action_delete_image -> {
+                lifecycleScope.launch {
+                    val file = File(checkNotNull(intent.getStringExtra(EXTRA_IMAGE_NAME)))
```

Comment:
This could throw an exception, would be nice to check it outside of the File.

Reply:
True. I will add the catch and show appropriate message to user.

I have 1 doubt. @simtop  
_File_ constructor only throws _NullPointerException_. And _checkNotNull_ throws  _IllegalStateException_ . I dont think _NullPointerException_ will ever be thrown when _checkNotNull_ exists.

Should I :-
1. Remove _checkNotNull_ and check for _File_'s Exception ? OR
2. Check for _checkNotNull_'s exception ?

Reply:
I have made the changes. Please feel free to open the conversation for any improvements.